include/deal.II/matrix_free/matrix_free.templates.h

```diff
@@ -1474,6 +1475,231 @@ MatrixFree<dim, Number, VectorizedArrayType>::initialize_indices(
   for (auto &di : dof_info)
     di.compute_vector_zero_access_pattern(task_info, face_info.faces);
 
+#ifdef DEAL_II_WITH_MPI
+  {
+    // non-buffering mode is only supported if the indices of all cells are
+    // of all dof_info are contiguous.
+    bool is_non_buffering_sm_supported = true;
+    for (const auto &di : dof_info)
+      for (const auto &v : di.index_storage_variants[2])
+        is_non_buffering_sm_supported &=
+          (v == internal::MatrixFreeFunctions::DoFInfo::IndexStorageVariants::
+                  contiguous);
+
+    is_non_buffering_sm_supported =
+      Utilities::MPI::min(static_cast<unsigned int>(
+                            is_non_buffering_sm_supported),
+                          task_info.communicator);
+
+    const MPI_Comm communicator_sm = this->task_info.communicator_sm;
+
+    if (is_non_buffering_sm_supported)
+      {
+        // TODO: move into Utilities::MPI
```

Comment:
Do you intend to work on this TODO here or elsewhere?

Reply:
I would like to move this - in a follow-up PR - to `Utilities::MPI`. Does this make sense or is this function too specialized? I am using practically these lines also in #11098.

Reply:
I think such a function would be of general use, so the `MPI` namespace seems right.